integration/simulation/simulation_azure_enclaves_test.go

```diff
@@ -45,10 +50,14 @@ func TestOneAzureEnclaveNodesMonteCarloSimulation(t *testing.T) {
 	simParams.AvgNetworkLatency = simParams.AvgBlockDuration / 15
 	simParams.AvgGossipPeriod = simParams.AvgBlockDuration / 3
 
+	if len(vmIPs) > simParams.NumberOfNodes {
```

Comment:
Previously, there was a warning in a TODO to not accidentally do this. I replaced it with a proper check.